### DISCUSSION TITLE
Dynamically resizing containers with mouse events?

### DISCUSSION THREAD

**Discussion by bradlarsen** [Textualize/textual#4230] [Q&A] [answered]
Is there a way with Textual layout to have the window split into two panes, left and right, and allow resizing the widths of the two panes by clicking with the mouse and dragging?

That is, I'm looking to add similar functionality to a Textual app that you get from Vim splits or tmux panes, where you can easily resize them with the mouse.

**Comment by davep** [comment, marked as answer]
There isn't anything built into Textual that directly supports this. I have seen at least one application built with Textual where the author has made this a feature though (sadly I can't remember which, or where I saw it -- perhaps here in GitHub discussions or it might have been mentioned in the #show-and-tell channel on the Textualize Discord server).

**Reply by TomJGooding** [reply]
@davep You might be thinking of [NoteSH](https://github.com/Cvaniak/NoteSH?tab=readme-ov-file#--you-can-resize-notes)? I remember this was one of the first apps I saw built using Textual and thinking "you can do _that_ in a terminal?!".

**Reply by davep** [reply]
@TomJGooding Perhaps, although I remember that as been a windowing app rather than a tiled app.

**Reply by bradlarsen** [reply]
Thank you both for the pointers — I will take a closer look!

**Comment by RyannDaGreat** [comment]
If someone makes a draggable pane (the kind in Tmux) - please send a pull request - this is a feature I REALLY REALLY want lol

**Comment by Lauszus** [comment]
Please see the discussion at: https://github.com/Textualize/textual/discussions/4834
